Examples/Algorithms/TrackFinding/include/ActsExamples/TrackFinding/AdaptiveHoughTransformSeeder.hpp

```diff
@@ -0,0 +1,478 @@
+ public:
+  AccumulatorSection() = default;
+
+  AccumulatorSection(float xw, float yw, float xBegin, float yBegin,
+                     int div = 0, const std::vector<unsigned> &indices = {},
+                     const std::vector<float> &history = {});
+
+  /// @brief keep indices and update parameters of the box
+  /// This method is useful when changing direction of the search
+  void updateDimensions(float xw, float yw, float xBegin, float yBegin);
+
+  /// @brief keep indices and update parameters of the box by scalling
+  /// @param xs - scale in x direction, if bigger than 1 the size increases
+  /// @param ys - scale in y direction
+  /// The box is recentred
+  void expand(float xs, float ys);
+
+  inline unsigned count() const {
+    return static_cast<unsigned>(m_indices.size());
+  }
+  inline const std::vector<unsigned> &indices() const { return m_indices; }
+  inline std::vector<unsigned> &indices() { return m_indices; }
+
+  // create section that is bottom part this this one
+  // +------+
+  // |      |
+  // +------+
+  // |     <|-- this part
+  // +------+
+  AccumulatorSection bottom(float yFraction = 0.5) const;
+  // see @bottom
+  AccumulatorSection top(float yFraction = 0.5) const;
+  // @see @bottom
+  AccumulatorSection left(float xFraction = 0.5) const;
+  // @see @bottom
+  AccumulatorSection right(float xFraction = 0.5) const;
+
+  // create section that is bottom left corner of this this one
+  // by default the section is divided into 4 quadrants,
+  // if parameters are provided the quadrants size can be adjusted
+  // +---+---+
+  // |   |   |
+  // +---+---+
+  // |   |  <|-- this part
+  // +---+---+
+  AccumulatorSection bottomRight(float xFraction = 0.5,
+                                 float yFraction = 0.5) const;
+  AccumulatorSection bottomLeft(float xFraction = 0.5,
+                                float yFraction = 0.5) const;
+
+  AccumulatorSection topLeft(float xFraction = 0.5,
+                             float yFraction = 0.5) const;
+  AccumulatorSection topRight(float xFraction = 0.5,
+                              float yFraction = 0.5) const;
+
+  /// @brief true if the line defined by given parameters passes the section
+  /// @param function is callable used to check crossing at the edges
+  template <typename F>
+  inline bool isLineInside(F function) const {
+    const float yB = function(m_xBegin);
+    const float yE = function(m_xBegin + m_xSize);
+    return (yE > yB) ? yB < m_yBegin + m_ySize && yE > m_yBegin
+                     : yB > m_yBegin && yE < m_yBegin + m_ySize;
+  }
+
+  /// @brief check if the lines cross inside the section
+  /// @brief line1 - functional form of line 1
+  /// @brief line1 - functional form of line 2
+  /// @warning note that this function is assuming that these are lines
+  /// It may be incorrect assumption for rapidly changing function or large
+  /// sections
+  /// @return true if the lines cross in the section
+  template <typename F>
```

Comment:
Same here. The added benefit is that you document the call signature of `F`.

Reply:
added